WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentsAlertPresenterAdaptor.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import Combine
+
+final class CardPresentPaymentsAlertPresenterAdaptor: CardPresentPaymentAlertsPresenting {
+    let paymentAlertPublisher: AnyPublisher<CardPresentPaymentEvent, Never>
+
+    private let paymentAlertSubject: PassthroughSubject<CardPresentPaymentEvent, Never> = PassthroughSubject()
+
+    private var latestReaderConnectionHandler: ((String) -> Void)?
+
+    init() {
+        paymentAlertPublisher = paymentAlertSubject.eraseToAnyPublisher()
+    }
+
+    func present(viewModel: CardPresentPaymentsModalViewModel) {
+        paymentAlertSubject.send(.showAlert(viewModel))
+    }
+
+    func foundSeveralReaders(readerIDs: [String], connect: @escaping (String) -> Void, cancelSearch: @escaping () -> Void) {
+        let wrappedConnectionHandler = { [weak self] readerID in
+            connect(readerID)
+            self?.latestReaderConnectionHandler = nil
+        }
+        self.latestReaderConnectionHandler = wrappedConnectionHandler
+        paymentAlertSubject.send(.showReaderList(readerIDs, selectionHandler: wrappedConnectionHandler))
+    }
+
+    func updateSeveralReadersList(readerIDs: [String]) {
```

Comment:
Thought for the future: from a previous look at these two states in `CardReaderConnectionController`, `updateSeveralReadersList` state is more for the existing app UI implementation detail to update the reader list

https://github.com/woocommerce/woocommerce-ios/blob/d67907e240714a0a866d148faf9069de89d61f9d/WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift#L334-L338

If we ever refactor the card reader connection UI states, we probably just need one state with the card reader list.

Reply:
Yeah, we only have one state in the new UI – we have to implement both these functions, but both get adapted to the single `showReaderList` event.